### ISSUE TITLE
Deep Learning CNN Model for Tomato Leaf Disease Prediction

### ISSUE THREAD

**Issue by keshripritesh** [omroy07/AgriTech#260]
# 🌱 New Project: Deep Learning CNN Model for Tomato Leaf Disease Prediction

## 📌 Description

This project introduces a **Deep Learning-based Convolutional Neural Network (CNN)** model trained on a **Tomato Leaf Image Dataset** to detect and classify various tomato plant diseases. The model can distinguish between multiple disease categories as well as healthy leaves, helping farmers and researchers in early disease detection and crop protection.

---

## 🦠 Diseases Covered

The trained CNN model can accurately classify tomato leaves into the following categories:

* Bacterial spot
* Early blight
* Late blight
* Leaf mold
* Septoria leaf spot
* Spider mites (Two-spotted spider mite)
* Target spot
* Tomato Yellow Leaf Curl Virus
* **Healthy Tomato Leaf** 🌿

---

## 📊 Model Performance

* ✅ Training Accuracy: **99.96%**
* ✅ Testing Accuracy: **96.99%**
* ⚖️ The model is **not overfit or underfit**, indicating strong generalization.

---

## 🖼️ Sample Outputs

* Healthy Tomato Leaf

<img width="815" height="740" alt="Image" src="https://github.com/user-attachments/assets/f2d258cd-c72e-4a14-bcc5-ac1f571dd8b0" />

* Tomato Leaf with Bacterial Spot

<img width="815" height="740" alt="Image" src="https://github.com/user-attachments/assets/fa0832ec-248f-4b8e-9986-8b9a0923df6d" />

* Tomato Leaf with Septoria Leaf Spot

<img width="815" height="740" alt="Image" src="https://github.com/user-attachments/assets/f3b98275-822c-4db7-87d7-74f1481a5dc0" />

* Tomato Leaf with Tomato Yellow Leaf Curl Virus

<img width="815" height="740" alt="Image" src="https://github.com/user-attachments/assets/ebc35fb5-0ec6-49c4-9231-a36c3b9c5c93" />

---

## 📂 Repo Placement

```
projects/Day-XXX-Tomato-Disease-Prediction-CNN/
│── dataset/
│── model/
│── tomato_cnn.py
│── README.md
```

---

## ⚙️ Tech Stack

* **Python 3**
* **TensorFlow / Keras** (Deep Learning)
* **OpenCV / Pillow** (Image Preprocessing)
* **NumPy, Pandas** (Data Handling)
* **Matplotlib / Seaborn** (Visualization)

---

## ▶️ How to Run

1. Clone the repository
2. Navigate to the project folder
3. Install dependencies:

   ```bash
   pip install tensorflow keras numpy pandas matplotlib seaborn opencv-python
   ```
4. Train or load the pretrained model:

   ```bash
   python tomato_cnn.py
   ```
5. Test on sample images

---

## 🚀 Future Enhancements

* Deploy as a **Streamlit web app** for user-friendly predictions
* Add **real-time detection** using camera feed
* Extend the dataset with more crop varieties
* Provide **disease treatment recommendations** along with predictions

---

🔗 **Event:** GirlScript Summer of Code (GSSoC’25)
🏷️ **Type:** New Project Addition

---

**Comment by omroy07** [comment]
Assigned
